Mathlib/Topology/Algebra/Algebra.lean

```diff
@@ -74,16 +78,471 @@ theorem algebraMapCLM_toLinearMap : (algebraMapCLM R A).toLinearMap = Algebra.li
   rfl
 #align algebra_map_clm_to_linear_map algebraMapCLM_toLinearMap
 
+/-- If `R` is a discrete topological ring, then any topological ring `S` which is an `R`-algebra
+  is also a topological `R`-algebra. -/
+instance DiscreteTopology.continuousSMul [TopologicalSemiring A] [DiscreteTopology R] :
+    ContinuousSMul R A := continuousSMul_of_algebraMap _ _ continuous_of_discreteTopology
+
 end TopologicalAlgebra
 
-section TopologicalAlgebra
+section TopologicalAlgebra --
+section
 
 variable {R : Type*} [CommSemiring R]
 variable {A : Type u} [TopologicalSpace A]
 variable [Semiring A] [Algebra R A]
 
 #align subalgebra.has_continuous_smul SMulMemClass.continuousSMul
 
+end
+section
+
+variable (R : Type*) [CommSemiring R] [TopologicalSpace R] [TopologicalSemiring R]
+  (A : Type*) [Semiring A] [TopologicalSpace A] [TopologicalSemiring A]
+
+/-- Continuous algebra homomorphisms between algebras. We only put the type classes that are
+  necessary for the definition, although in applications `M` and `B` will be topological algebras
+  over the topological ring `R`. -/
```

Comment:
Style comment here and throughout. I won't mark the other ones, but please find them.
```suggestion
necessary for the definition, although in applications `M` and `B` will be topological algebras
over the topological ring `R`. -/
```